hazelcast/src/test/java/com/hazelcast/test/starter/test/AddressConstructorTest.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test.starter.test;
+
+import com.hazelcast.nio.Address;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelTest;
+import com.hazelcast.test.annotation.QuickTest;
+import com.hazelcast.test.starter.AddressConstructor;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class AddressConstructorTest {
+
+    @Test
+    public void addressCloneTest() throws Exception {
+        Address address = new Address("172.16.16.1", 4223);
+
+        AddressConstructor constructor = new AddressConstructor(Address.class);
+        Address clonedAddress = (Address) constructor.createNew(address);
+
+        assertEquals(clonedAddress.getHost(), address.getHost());
```

Comment:
Shouldn't the order of the parameters be the other way? Since first argument is "expected", so this might mislead you when you see the assertion error.

Reply:
Oh, yes. Fixed that also in the `ConfigConstructorTest`.